src/topology/bases.lean

```diff
@@ -9,10 +9,40 @@ import topology.continuous_on
 /-!
 # Bases of topologies. Countability axioms.
 
+A topological basis on a topological space `t` is a collection of sets,
+such that taking the union of these sets generates all open sets, without the need to take
+finite intersections of them. This file introduces a framework for dealing with these collections,
+and also what more we can say under certain countability conditions on bases,
+which are referred to as first- and second-countable.
+We also briefly cover the theory of separable spaces, which are those with a countable, dense
+subset. If a space is second-countable, and also has a countably generated uniformity filter
+(for example, if `t` is a metric space), it will automatically be separable (and indeed, these
+conditions are equivalent in this case).
+
+## Main definitions
+
+* `is_topological_basis s`: Indicates whether the topology `t` is generated by `s`, but only
+  allowing unions of sets in `s`.
```

Comment:
```suggestion
* `is_topological_basis s`: Indicates whether the topology `t` is generated by `s` by taking unions.
```
(I think the "but" here is just a distraction, as an alternative hasn't even been raised.)

Reply:
Moreover, I don't like the "Indicates whether". It's not that this typeclass has information either way: it says it is! Perhaps just "The topological `t` is ...".